app/validators/content_item_uniqueness_validator.rb

```diff
@@ -29,16 +30,15 @@ def validate(record)
     base_path = location.base_path
     user_version = user_facing_version.number
 
-    matching_items = ContentItemFilter.filter(
+    other_content_items = ContentItem.where("content_items.id <> #{content_item.id}")
+    matching_items = ContentItemFilter.new(scope: other_content_items).filter(
       state: state_name,
       locale: locale,
       base_path: base_path,
       user_version: user_version,
     )
 
-    additional_items = matching_items - [content_item]
-
-    if additional_items.any?
+    if matching_items.any?
```

Comment:
I wouldn't call that a micro optimisation if it were the case, but yes this is a rare example of AR doing the right thing :)